extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java

```diff
@@ -16,12 +16,23 @@
 
 import de.sovity.edc.ext.wrapper.api.ui.model.TransferHistoryPage;
 import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
 
 @RequiredArgsConstructor
 public class TransferHistoryPageApiService {
 
+    private final TransferHistoryPageDataFetcher transferHistoryPageDataFetcher;
+
+    private final TransferProcessAssetFetcher transferProcessAssetFetcher;
+
     public TransferHistoryPage transferHistoryPage() {
-        // TODO: Implement
-        throw new IllegalStateException("Not yet implemented");
+
+        var transferHistoryPageEntries = transferHistoryPageDataFetcher.getTransferHistoryEntries();
+        return new TransferHistoryPage(transferHistoryPageEntries);
+    }
+
+    public Asset fetchAssetForTransferProcess(String transferProcessId) {
+
+        return transferProcessAssetFetcher.getAssetForTransferHistoryPage(transferProcessId);
     }
```

Comment:
uhm, both these methods basically only call the other function, so you could have a

`TransferHistoryPageApiService` which should contain what your datafetcher currently does.

`TransferProcessGetAssetApiService` which contains the content of your transfer process asset fetcher.